lib/abci/handlers/endBlockHandlerFactory.js

```diff
@@ -68,23 +73,40 @@ function endBlockHandlerFactory(
     const header = blockExecutionContext.getHeader();
     const coreChainLock = latestCoreChainLock.getChainLock();
 
-    let response = {};
+    let validatorSetUpdate;
+    if (await validatorQuorums.rotate(height, coreChainLock.height, header.lastCommitHash)) {
```

Comment:
`lastCommitHash` can be manipulated. As I remember we planned to use threshold signature as a seed. I would recommend making an RPC call to get a commit on begin block and put it to block execution context. Having it in block execution context allows us to use it here and also in query handlers to provide app hash signature for proofs.

Reply:
good catch! I was using lastCommitHash just temporarily until getting a better option. Happy to use the threshold signatures now that we have them.  js-abci is updated https://github.com/dashevo/js-abci/pull/3